File: sample/src/main/java/contacts/sample/cheatsheet/blockednumbers/java/QueryBlockedNumbersActivity.java
package contacts.sample.cheatsheet.blockednumbers.java;

import static contacts.core.WhereKt.*;

import android.app.Activity;

import java.util.List;

import contacts.core.*;
import contacts.core.entities.BlockedNumber;

public class QueryBlockedNumbersActivity extends Activity {

    List<BlockedNumber> getAllBlockedNumbers() {
        return ContactsFactory.create(this).blockedNumbers().query().find();
    }

    List<BlockedNumber> getBlockedNumbersContainingNumber(String number) {
        return ContactsFactory.create(this)
                .blockedNumbers()
                .query()
                .where(
                        or(
                                contains(BlockedNumbersFields.Number, number),
                                contains(BlockedNumbersFields.NormalizedNumber, number)
                        )
                )
                .find();
    }
}

File: sample/src/main/java/contacts/sample/cheatsheet/blockednumbers/kotlin/QueryBlockedNumbersActivity.kt
package contacts.sample.cheatsheet.blockednumbers.kotlin

import android.app.Activity
import contacts.core.*
import contacts.core.entities.BlockedNumber

class QueryBlockedNumbersActivity : Activity() {

    fun getAllBlockedNumbers(): List<BlockedNumber> =
        Contacts(this).blockedNumbers().query().find()

    fun getBlockedNumbersContainingNumber(number: String): List<BlockedNumber> = Contacts(this)
        .blockedNumbers()
        .query()
        .where { (Number contains number) or (NormalizedNumber contains number) }
        .find()
}